les/server_handler.go

```diff
@@ -223,648 +221,109 @@ func (h *serverHandler) handleMsg(p *clientPeer, wg *sync.WaitGroup) error {
 	}
 	defer msg.Discard()
 
-	var (
-		maxCost uint64
-		task    *servingTask
-	)
 	p.responseCount++
 	responseCount := p.responseCount
-	// accept returns an indicator whether the request can be served.
-	// If so, deduct the max cost from the flow control buffer.
-	accept := func(reqID, reqCnt, maxCnt uint64) bool {
-		// Short circuit if the peer is already frozen or the request is invalid.
-		inSizeCost := h.server.costTracker.realCost(0, msg.Size, 0)
```

Comment:
Would be super super nice if we can encapsulate these "before-serving" logic and "after-serving" logic into two separate functions: "prepare"  and "finalize". The execution is in the middle.

Just throw out the thoughts here. We can open the following PR if you also think it's a good idea.

Reply:
Sure, let's do it in a follow-up PR.